opm/common/data/SimulationDataContainer.hpp

```diff
@@ -86,6 +95,12 @@ namespace Opm {
         const std::vector<double>& facepressure() const;
         const std::vector<double>& faceflux    () const;
 
+
+        std::unordered_map<std::string, std::vector<double>>::iterator celldata_begin();
+        std::unordered_map<std::string, std::vector<double>>::iterator celldata_end();
+        const std::unordered_map<std::string, std::vector<double>>::const_iterator celldata_cbegin() const;
+        const std::unordered_map<std::string, std::vector<double>>::const_iterator celldata_cend() const;
```

Comment:
> This should probably be a const_iterator rather than a const const::iterator.

Well - currently it is `const xxx const_iterator`- so somewhere in between. I must admit I was quite uncertain on how to sprinkle the `const` around. What I want is an iterator which acts like:
```
for (auto iter = container.begin(); iter != container.end(); ++iter) {
     // Now iter should point to a *constant* object which can not be modified in calling scope.
}
```
I actually think (but will not even consider to insist ...) that I had it right.

> having a few type aliases in place 

OK; must admit I have a bit mixed feelings on these *simplify-the-declarations* type aliases; calling scope can hopefully get away with using `auto` - but otherwise we need to know the return type, and then these aliases just make the code more opaque for downstream consumers - I'm not hell-bent on this.